xhr/send-content-type-charset.htm

```diff
@@ -70,14 +75,30 @@
       )
       request(
         "text/x-pink-unicorn; charset=windows-1252; charset=bogus; notrelated; charset=ascii",
-        "text/x-pink-unicorn; charset=UTF-8; charset=UTF-8; notrelated; charset=UTF-8",
-        "If multiple charset parameters are given, all should be rewritten"
+        "text/x-pink-unicorn;charset=UTF-8",
+        "Multiple charset parameters deduplicate, bogus parameter dropped"
       )
       request(
         null,
         "text/plain;charset=UTF-8",
         "No content type set, give MIME and charset"
       )
+      request(
+        "text/plain;charset= utf-8",
+        "text/plain;charset=UTF-8",
+        "charset with space")
+      request(
+        "text/plain;charset=\"utf-8\"",
+        "text/plain;charset=UTF-8",
+        "charset in double quotes")
+      request(
+        "text/plain;charset=\" utf-8\"",
+        "text/plain;charset=UTF-8",
+        "charset in double quotes with space")
+      request(
+        "text/plain;charset=\"u\\t\\f-8\"",
+        "text/plain;charset=UTF-8",
+        "charset in double quotes with backslashes")
     </script>
```

Comment:
Can you add tests that non-charset parameters also get preserved (and parsed/re-serialized), when a charset parameter is present?